src/python/pants/backend/codegen/thrift/apache/java/rules.py

```diff
@@ -60,12 +72,78 @@ class InjectApacheThriftJavaDependencies(InjectDependenciesRequest):
     inject_for = ThriftDependenciesField
 
 
+@dataclass(frozen=True)
+class ApacheThriftJavaRuntimeForResolveRequest:
+    resolve_name: str
+
+
+@dataclass(frozen=True)
+class ApacheThriftJavaRuntimeForResolve:
+    address: Address
+
+
+_LIBTHRIFT_GROUP = "org.apache.thrift"
+_LIBTHRIFT_ARTIFACT = "libthrift"
+
+
+@rule
+async def resolve_apache_thrift_java_runtime_for_resolve(
+    request: ApacheThriftJavaRuntimeForResolveRequest,
+    jvm_artifact_targets: AllJvmArtifactTargets,
+    jvm: JvmSubsystem,
+) -> ApacheThriftJavaRuntimeForResolve:
+    try:
+        addresses = find_jvm_artifacts_or_raise(
+            required_coordinates=[
+                UnversionedCoordinate(
+                    group=_LIBTHRIFT_GROUP,
+                    artifact=_LIBTHRIFT_ARTIFACT,
+                )
+            ],
+            resolve=request.resolve_name,
+            jvm_artifact_targets=jvm_artifact_targets,
+            jvm=jvm,
+        )
+        return ApacheThriftJavaRuntimeForResolve(list(addresses)[0])
```

Comment:
Should this error if there is greater than one target?

Reply:
It won't since it relies on an implementation assumption that one artifact coordinate maps to exactly one `jvm_artifact` address. I didn't want to start changing the dependency inference types here to start supporting multiple addresses. I will eventually but seemed too much like churn at the time, although now it has caused reader confusion.